src/map/status.c

```diff
@@ -13397,25 +13397,127 @@ static bool status_readdb_sizefix(char *fields[], int columns, int current)
+	return true;
+}
+
+static bool status_read_scdb_libconfig_sub(struct config_setting_t *it, int idx, const char *source)
+{
+	nullpo_retr(false, it);
+	nullpo_retr(false, source);
+
+	int i32;
+	const char *name = config_setting_name(it);
+
+	if (!script->get_constant(name, &i32) || i32 < SC_NONE || i32 >= SC_MAX) {
+		ShowWarning("status_read_scdb_libconfig_sub: Invalid status type (%s) in \"%s\" entry #%d, skipping.\n", name, source, idx);
+		return false;
+	}
+
+	struct config_setting_t *fg = libconfig->setting_get_member(it, "Flags");
+	status->read_scdb_libconfig_sub_flag(fg, i32, source);
+
+	return true;
+}
+
+static bool status_read_scdb_libconfig_sub_flag(struct config_setting_t *it, int type, const char *source)
+{
+	nullpo_retr(false, it);
+	nullpo_retr(false, source);
+	Assert_retr(false, type >= SC_NONE && type < SC_MAX);
+
+	int i = 0;
+	struct config_setting_t *t = NULL;
+	while ((t = libconfig->setting_get_elem(it, i++)) != NULL) {
+		const char *flag = config_setting_name(t);
+		bool on = libconfig->setting_get_bool_real(t);
+
+		if (strcmpi(flag, "NoDeathReset") == 0) {
+			if (on) {
+				status->dbs->sc_conf[type] |= SC_NO_REM_DEATH;
+			} else {
+				status->dbs->sc_conf[type] &= ~SC_NO_REM_DEATH;
+			}
+		} else if (strcmpi(flag, "NoSave") == 0) {
+			if (on) {
+				status->dbs->sc_conf[type] |= SC_NO_SAVE;
+			} else {
+				status->dbs->sc_conf[type] &= ~SC_NO_SAVE;
+			}
+		} else if (strcmpi(flag, "NoDispelReset") == 0) {
+			if (on) {
+				status->dbs->sc_conf[type] |= SC_NO_DISPELL;
+			} else {
+				status->dbs->sc_conf[type] &= ~SC_NO_DISPELL;
+			}
+		} else if (strcmpi(flag, "NoClearanceReset") == 0) {
+			if (on) {
+				status->dbs->sc_conf[type] |= SC_NO_CLEARANCE;
+			} else {
+				status->dbs->sc_conf[type] &= ~SC_NO_CLEARANCE;
+			}
+		} else if (strcmpi(flag, "Buff") == 0) {
+			if (on) {
+				status->dbs->sc_conf[type] |= SC_BUFF;
+			} else {
+				status->dbs->sc_conf[type] &= ~SC_BUFF;
+			}
+		} else if (strcmpi(flag, "Debuff") == 0) {
+			if (on) {
+				status->dbs->sc_conf[type] |= SC_DEBUFF;
+			} else {
+				status->dbs->sc_conf[type] &= ~SC_DEBUFF;
+			}
+		} else if (strcmpi(flag, "NoMadoReset") == 0) {
+			if (on) {
+				status->dbs->sc_conf[type] |= SC_MADO_NO_RESET;
+			} else {
+				status->dbs->sc_conf[type] &= ~SC_MADO_NO_RESET;
+			}
+		} else if (strcmpi(flag, "NoAllReset") == 0) {
+			if (on) {
+				status->dbs->sc_conf[type] |= SC_NO_CLEAR;
+			} else {
+				status->dbs->sc_conf[type] &= ~SC_NO_CLEAR;
+			}
+		} else if (strcmpi(flag, "Visible") == 0) {
+			if (on) {
+				status->dbs->DisplayType[type] = true;
+			} else {
+				status->dbs->DisplayType[type] = false;
+			}
+		} else {
+			status->read_scdb_libconfig_sub_flag_additional(it, type, source);
+		}
+	}
 	return true;
 }
+
+static void status_read_scdb_libconfig_sub_flag_additional(struct config_setting_t *it, int type, const char *source)
```

Comment:
change this structure to be bool and default return of false, plugins shall change this to true, otherwise main function returns a warning.